### ISSUE TITLE
Failed to transcode QR codes pointing to localhost URLs

### ISSUE THREAD

**Issue by plmr-org** [monyxie/qr-lite#33]
Currently writing an application that generates QR codes as BLOBs in the back-end and sends them to the front-end where they are displayed. Here's the QR code in question:
![image](https://github.com/user-attachments/assets/fa02c709-ee8d-48fe-a225-75566434eccc)

It should be pointing towards `http://localhost:3000/<base64>` or something of that nature. The QR code works through my Android phone's built-in QR code scanner, though not through the extension.

**Comment by plmr-org** [comment]
Strangely there are also no error messages raised by the extension. Here's what I received which may be more useful:
![2024-07-30_14-24](https://github.com/user-attachments/assets/e319c25f-d72c-4946-8393-1117c79107d1)

**Comment by plmr-org** [comment]
Strangely there are also no error messages raised by the extension. Here's what I received which may be more useful:
![2024-07-30_14-24](https://github.com/user-attachments/assets/e319c25f-d72c-4946-8393-1117c79107d1)

**Comment by plmr-org** [comment]
This also happens on Firefox by the way:
![2024-07-30_14-25](https://github.com/user-attachments/assets/b7d2385c-f400-494e-9579-b23221ee7ae9)
<sub>Sorry for the comment spam.</sub>

**Comment by monyxie** [comment]
Hi, thanks for the bug report. However I can't seem to reproduce this on my side. Can you provide the HTML source for the web page where this issue happened?

**Comment by plmr-org** [comment]
It effectively boils down to this:
```html
<div className="h-80 w-80">
    <img
        alt="QR Code"
        height={450}
        src={qrCode}
        width={450}
    />
</div>
```
I'm now wondering if the image being encapsulated in a `div` is what could be causing this issue.

If this doesn't suffice, I plan to publicise the page anyway for diagnostic purposes if that's any easier for you.

**Comment by plmr-org** [comment]
Weirdly, region scanning works fine, the option to scan the QR code in the image is what's problematic as shown here:
![image](https://github.com/user-attachments/assets/06672167-bc9b-472a-ae25-fe20f3a7c080)

**Comment by monyxie** [comment]
Yes, I suspect that this is a image loading issue. It would be nice if you could provide the relevant parts of the rendered HTML code (not your framework code). I'm specifically interested in these info:
- the image format (jpg or png or else?)
- the way the image is loaded (http or data uri or else?)
- additionally, if it's loaded via http, the request dump (HAR file) of the image request

**Comment by monyxie** [comment]
This is likely due to the image's src attribute being a `blob:` URL, which can not be used across different browsing contexts, causing the extension to fail to load the image.
The upcoming v3.3.0 release will change how the extension retrieves the image data and will therefore make unrevoked `blob:` URLs work. However revoked `blob:` URLs will still fail due to Firefox not providing such URL in its context menu API .

**Comment by monyxie** [comment]
> However revoked `blob:` URLs will still fail due to Firefox not providing such URL in its context menu API .

Nevermind this. Images loaded via a revoked `blob:` URL can be retrieved using another method.

This bug will be fully fixed in the next release.
